frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java

```diff
@@ -169,11 +174,26 @@ public void initialize() {
 
                     getDataCenter().setIsChangeable(isChangeable);
                     if (!isChangeable) {
-                        // Set the selected item on the storage domains list.
-                        AsyncDataProvider.getInstance().getStorageDomainById(
-                                new AsyncQuery<>(storageDomain -> {
-                                    getStorageDomain().setItems(Collections.singletonList(storageDomain));
-                                }), limitToStorageDomainId);
+                        // Set the selected item on the storage domains and data centers lists.
+                        Frontend.getInstance().runQuery(
+                                QueryType.GetStorageDomainListById,
+                                new IdQueryParameters(limitToStorageDomainId),
+                                new AsyncQuery<QueryReturnValue>(returnValue -> {
+                            ArrayList<StorageDomain> domains = returnValue.getReturnValue();
+                            if (!domains.isEmpty()) {
+                                StorageDomain storageDomain = domains.get(0);
+                                // Set the selected item on the storage domains list.
+                                getStorageDomain().setItems(Collections.singletonList(storageDomain));
+                                Guid dcId = storageDomain.getStoragePoolId();
```

Comment:
And the result also contains the `storagePoolId` field, so we can use it to display proper Data Center in the appropriate field of the "Upload Image" Dialog.